spec/schema_tools/hash_spec.rb

```diff
@@ -217,6 +225,27 @@ class NestedObjectNoProperties
       hash['person']['first_name'].should == 'Pit'
     end
 
+    it 'has nested anyOf type object ' do
```

Comment:
It's not clear from these specs how `AnyOfDefinition` behaves differently from `AllOfDefinition`.